src/components/NcUserStatusIcon/NcUserStatusIcon.vue

```diff
@@ -30,124 +30,85 @@ This component displays a user status icon.
 </docs>
 
 <template>
-	<span v-if="activeStatus"
+	<span v-if="status"
 		class="user-status-icon"
 		:class="{
-			'user-status-icon--invisible': ['invisible', 'offline'].includes(status),
+			'user-status-icon--invisible': isInvisible,
 		}"
+		:aria-hidden="!ariaLabel || undefined"
+		:aria-label
 		role="img"
-		:aria-hidden="ariaHidden"
-		:aria-label="ariaLabel"
 		v-html="activeSvg" /> <!-- eslint-disable-line vue/no-v-html -->
 </template>
 
-<script>
+<script setup lang="ts">
 import axios from '@nextcloud/axios'
 import { generateOcsUrl } from '@nextcloud/router'
 import { getCapabilities } from '@nextcloud/capabilities'
+import { computed, watch } from 'vue'
+import { getUserStatusText } from '../../utils/UserStatus.ts'
+import { t } from '../../l10n.js'
 
 import onlineSvg from '../../assets/status-icons/user-status-online.svg?raw'
 import awaySvg from '../../assets/status-icons/user-status-away.svg?raw'
 import dndSvg from '../../assets/status-icons/user-status-dnd.svg?raw'
 import invisibleSvg from '../../assets/status-icons/user-status-invisible.svg?raw'
-
-import { getUserStatusText } from '../../utils/UserStatus.ts'
-import { t } from '../../l10n.js'
-
-export default {
-	name: 'NcUserStatusIcon',
-
-	props: {
-		/**
-		 * Set the user id to fetch the status
-		 */
-		user: {
-			type: String,
-			default: null,
-		},
-
-		/**
-		 * Set the status
-		 *
-		 * @type {'online' | 'away' | 'busy' | 'dnd' | 'invisible' | 'offline'}
-		 */
-		status: {
-			type: String,
-			default: null,
-			validator: (value) => [
-				'online',
-				'away',
-				'busy',
-				'dnd',
-				'invisible',
-				'offline',
-			].includes(value),
-		},
-
-		/**
-		 * Set the `aria-hidden` attribute
-		 *
-		 * @type {'true' | 'false'}
-		 */
-		ariaHidden: {
-			type: String,
-			default: null,
-			validator: (value) => [
-				'true',
-				'false',
-			].includes(value),
-		},
-	},
-
-	data() {
-		return {
-			fetchedUserStatus: null,
+import logger from '../../utils/logger.ts'
+
+const props = withDefaults(defineProps<{
+	/**
+	 * Set the user id to fetch the status
+	 */
+	user?: string,
+
+	/**
+	 * The user preloaded user status.
+	 */
+	status?: 'online' | 'away' | 'busy' | 'dnd' | 'invisible' | 'offline'
+
+	/**
+	 * Set the `aria-hidden` attribute
+	 */
+	ariaHidden?: boolean | 'true' | 'false'
+}>(), {
+	user: undefined,
+	status: undefined,
+	ariaHidden: false,
+})
+
+const status = defineModel<typeof props.status>('status', { default: null })
```

Comment:
Status is defined twice: once in `defineProps` and once in `defineModel`

The idea of `defineModel` macros is to replace corresponding items in `defineProps` + `defineEmits`.

Reply:
And the definition is different. Default value is `undefined` in one place and `null` in another

Reply:
good point, fixed